### PR TITLE
Support defining spinner keys and values within an options block

### PR DESCRIPTION
**Pull request by vincent-karuri**
Adds support for defining spinner keys and values within an options block e.g.:

```
{
        "key": "response_spinner_with_options",
        "openmrs_entity_parent": "",
        "openmrs_entity": "",
        "openmrs_entity_id": "",
        "type": "spinner",
        "hint": "Response Spinners",
        "options": [
          {
            "key": "yes",
            "text": "Yes",
            "value": false,
            "openmrs_entity": "",
            "openmrs_entity_id": ""
          },
          {
            "key": "no",
            "text": "No",
            "value": false,
            "openmrs_entity": "",
            "openmrs_entity_id": ""
          },
          {
            "key": "maybe",
            "text": "Maybe",
            "value": false,
            "openmrs_entity": "",
            "openmrs_entity_id": ""
          }
        ],
        "v_required": {
          "value": "true",
          "err": "Please enter response"
        },
        "openmrs_choice_ids": {
          "user_one": "1107AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA",
          "user_two": "1713AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"
        }
 }
```

### CODE REVIEW

File: sample/src/main/assets/json.form/basic_form.json [hunk 1]
```diff
@@ -353,6 +353,45 @@
...
+        "openmrs_entity_parent": "",
+        "openmrs_entity": "",
+        "openmrs_entity_id": "",
+        "type": "spinner",
+        "hint": "Response Spinners",
+        "options": [
+          {
+            "key": "yes",
+            "text": "Yes",
+            "value": false,
```
[the remark this answers is not in the document]
Comment (vincent-karuri) [reply sample/src/main/assets/json.form/basic_form.json:367]: Nothing about  how selections are saved has been altered. The only changes here are to pull the keys and values for the `options` block.

I also encourage testing this on the client apps to verify the same.

[the remark this answers is not in the document]
Comment (dubdabasoduba) [reply sample/src/main/assets/json.form/basic_form.json:367]: Tested on ANC.

[the remark this answers is not in the document]
Comment (dubdabasoduba) [reply sample/src/main/assets/json.form/basic_form.json:367]: I would recommend removing the `"value":true` attribute on the options. its only needed for checkboxes. Single value selections like radio buttons & the spinner don't need it.

File: sample/src/main/assets/json.form/basic_form.json [hunk 2]
```diff
@@ -353,6 +353,45 @@
...
+          }
+        ],
+        "v_required": {
+          "value": "true",
+          "err": "Please enter response"
+        },
+        "openmrs_choice_ids": {
+          "user_one": "1107AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA",
+          "user_two": "1713AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"
+        }
```
Comment (dubdabasoduba) [inline sample/src/main/assets/json.form/basic_form.json:393]: we should also update this functionality to make sure the openmrs_entities can be read from the options if the options are available. Then we wouldn't need this block for spinners with options

File: sample/src/main/assets/json.form/basic_form.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (vincent-karuri) [reply sample/src/main/assets/json.form/basic_form.json:367]: Done.

File: sample/src/main/assets/json.form/basic_form.json (hunk 2 above)
Comment (vincent-karuri) [reply sample/src/main/assets/json.form/basic_form.json:393]: Are these currently extracted in the widget factory?

Comment (vincent-karuri) [reply sample/src/main/assets/json.form/basic_form.json:393]: And what do the `openmrs_choice_ids` correspond to when compared with checkbox `options`?

Comment (dubdabasoduba) [reply sample/src/main/assets/json.form/basic_form.json:390]: They correspond to the key & the openmrs_entity_id

Comment (vincent-karuri) [reply sample/src/main/assets/json.form/basic_form.json:390]: Thanks.

Comment (vincent-karuri) [reply sample/src/main/assets/json.form/basic_form.json:390]: Done.

**Pull request merged**